### DISCUSSION TITLE
Ability to create collections from another collection and list them on the Home page

### DISCUSSION THREAD

**Discussion by kmatsuleva** [firecmsco/firecms#438] [Q&A] [answered]
**Expected result:** We have one main collection ("All Modules") in which we create other collections ("Module 1", "Module 2") with the required fields:

![image](https://user-images.githubusercontent.com/37826590/205015767-ba82ecc7-c9c5-45a5-a2ee-00e4d4524e3e.png)
![image](https://user-images.githubusercontent.com/37826590/205016022-7e6c21b7-8deb-49e6-892a-7bb7fd36cef4.png)

These collections, once created, are listed on the home page:

![image](https://user-images.githubusercontent.com/37826590/205017643-b81db0e1-88fc-4c77-8206-e2a72b08098b.png)

We can open them and fill in the fields with the necessary information:

![image](https://user-images.githubusercontent.com/37826590/205017683-7c0510a7-b2a0-49b1-8bee-5fd225785618.png)

**Actual:** The functionality from the first two screenshots is ready, but I'm having a hard time showing "Module 1" and "Module 2" on the home page.

**What I tried so far:**

- Entity callbacks with onFetch:
![image](https://user-images.githubusercontent.com/37826590/205022780-d85d501c-db9a-4aef-bb05-3832177c53a3.png)
- refresh state (failed because it's in onFetch)
- EntityCollectionsBuilder (got different and multiple errors) :
![image](https://user-images.githubusercontent.com/37826590/205023299-940afeac-6fe5-458d-a182-d9919b5f33ea.png)

Does anyone have any idea how and if it is possible to achieve this within the CMS? Any help would be appreciated.

**Comment by fgatti675** [comment, marked as answer]
Hi @kmatsuleva 
I have tried implementing this on my side and realised there were a couple of small pieces missing.
You can now access the datasource in the `EntityCollectionBuilder`, that will allow you to read data from Firestore upon creation of the CMS. I have also added a function that allows you to refresh the navigation data on demand.
I am writing an article on how to put all this together, but in the meantime, you can use this code as a template.
Remember to update to the latest `2.0.0-alpha.47` version!
```tsx
import React from "react";
import {
    buildCollection,
    EntityCollectionsBuilder,
    FirebaseCMSApp
} from "@camberi/firecms";

import "typeface-rubik";
import "@fontsource/ibm-plex-mono";

// TODO: Replace with your config
const firebaseConfig = {
    apiKey: "",
    authDomain: "",
    projectId: "",
    storageBucket: "",
    messagingSenderId: "",
    appId: ""
};

type Unit = {
    name: string;
    price: number;
}

const unitsCollection = buildCollection<Unit>({
    name: "Units",
    singularName: "Unit",
    group: "Main",
    path: "units",
    customId: true,
    callbacks: {
        onSaveSuccess: ({ context }) => {
            context.navigation.refreshNavigation();
        },
        onDelete: ({ context }) => {
            context.navigation.refreshNavigation();
        }
    },
    properties: {
        name: {
            name: "Name",
            validation: { required: true },
            dataType: "string"
        },
        price: {
            name: "Price",
            validation: {
                required: true,
                requiredMessage: "You must set a price between 0 and 1000",
                min: 0,
                max: 1000
            },
            description: "Price with range validation",
            dataType: "number"
        }
    }
});

export default function App() {

    const collectionBuilder: EntityCollectionsBuilder = async ({ dataSource }) => {
        const units = await dataSource.fetchCollection<Unit>({
            path: "units",
            collection: unitsCollection
        });
        const lessonCollections = units.map(unit => buildCollection({
            name: unit.values.name,
            path: `units/${unit.id}/lessons`,
            group: "Units",
            properties: {
                name: {
                    name: "Name",
                    dataType: "string"
                }
            }
        }));

        return [
            unitsCollection,
            ...lessonCollections
        ]
    };

    return <FirebaseCMSApp
        name={"My learning app"}
        collections={collectionBuilder}
        firebaseConfig={firebaseConfig}
    />;
}
```

**Comment by fgatti675** [comment]
Full explanation in https://firecms.co/docs/recipes/documents_as_subcollections ;)

**Comment by kmatsuleva** [comment]
This helped me a lot! Thank you very much @fgatti675 !
